### ISSUE TITLE
Follow up: fix remaining console nagging, when switching from 5 to 4

### ISSUE THREAD

**Issue by indigoxela** [backdrop-contrib/ckeditor5#43]
How to reproduce:

- Two filter formats, one has CKE4, one has CKE5
- CKE5 is the default (by weight)
- Create new node (open node form with 5 as editor)
- Inspect console

You'll find nagging by the CKEDITOR.resourceManager (v4!), that config.js and styles.js could not be found.

It seems like 4 initializes (not sure, what triggers it), before the actual "attach", where settings get applied. Instead it already initializes on page load. After switching from 5 to 4 once (settings applied), the error doesn't repeat on additional switching.

In #40 a workaround has been added, to provide a global var CKEDITOR_BASEPATH, to prevent CKE4 from "guessing" its own base path wrong, which prevents the editor from working even after switching to 4. But that only solves a part of the problem.

**Comment by indigoxela** [comment]
Note that in a new PR, the ckeditor library's file name is changed, which might make the previous workaround obsolete. Needs verification, though.

**Comment by quicksketch** [comment]
I'm still getting these errors even after the renaming of CKEditor 5's library to `ckeditor-dll.js` unfortunately.

**Comment by indigoxela** [comment]
> I'm still getting these errors

Yep, that's expected. Renaming only should make the [previous workaround](https://github.com/backdrop-contrib/ckeditor5/blob/1.x-1.x/js/ckeditor5.compat.js) obsolete.

The one thing I could never figure out: what triggers that? CKE4 isn't supposed to do anything, as CKE5 acts inside its own namespace... :thinking:

**Comment by quicksketch** [comment]
I'm taking a look at this issue.

**Comment by quicksketch** [comment]
Well instead of removing `ckeditor5.compat.js`, I repurposed it for a new work-around. It seems as though if CKEditor 4 is loaded onto a page but not initialized, it starts up its own internal instance with all default settings.

The PR at https://github.com/backdrop-contrib/ckeditor5/pull/71 updates some of those default settings to make it so that it does not look for a `config.js`, `styles.js`, or `styles.css` file, all of which are default files that do not ship with Backdrop core.

I'll make a corresponding core issue and PR, this fix should probably be directly in the core module and not in this one.

**Comment by quicksketch** [comment]
Now something very strange is happening. Both CKEditor 4 and 5 are being attached at the same time. By fixing the errors, the other CKEditor 4 instance successfully loads and weird floats over the top of CKEditor 5. Surprisingly, some of the buttons even work. :dizzy_face: 

![image](https://github.com/backdrop-contrib/ckeditor5/assets/100206/c2a22e8d-16f9-4618-a323-9d7fd2d6bbd0)

So I think we need a different solution. Rather than fixing the errors we need to find why this instance is being created and stop it.

**Comment by indigoxela** [comment]
Wow, that looks very strange. However, I can't reproduce it locally.

- Applied the patch
- Flushed caches
- Opened a node form (existing node)
- All good, no console nagging

Flushed caches several times, and re-opened the form. Still all good. Switching between 4 and 5 also doesn't cause any console nagging.

<s>There might be something specific on your setup, that triggers this odd behavior...</s>

EDIT: Arghhh, and right after submitting this comment ... it's reproducible here. :zany_face:

**Comment by indigoxela** [comment]
My current impression is, that this CKE4 instance behaves like a CKE5 balloon menu (sort of). Not sure, yet, if that's correct.

**Comment by quicksketch** [comment]
Ah, I found the real root. [CKEditor 4 _automatically_ attaches](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_inline.html#enabling-inline-editing) by default to any `contenteditable="true"` element on the page. When CKEditor 5 is loaded first, it creates a `contenteditable="true"` element, CKEditor 4 then sees it and automatically attaches its inline editor onto it too.

According to the CKEditor 4 docs, we can disable this with `CKEDITOR.disableAutoInline = false;`. I confirmed this works and updated my PR to just disable that option.
